armory/eval/evaluator.py

```diff
@@ -69,42 +69,50 @@ def __init__(
             self.manager = HostManagementInstance()
             return
 
-        # Download docker image on host
+        kwargs["image_name"] = self.ensure_image_present(image_name)
+        self.manager = ManagementInstance(**kwargs)
+
+    def ensure_image_present(self, image_name: str) -> str:
+        """if image_name is available, return it. Otherwise, pull it from dockerhub"""
         # TODO This seems like it still needs docker even in no docker mode
         #  we should fix this
-        log.info("Attempting to get Docker Client")
+
         docker_client = docker.from_env()
-        try:
-            docker_client.images.get(kwargs["image_name"])
-        except docker.errors.ImageNotFound:
-            log.info(f"Image {image_name} was not found. Downloading...")
+
+        # look first for  the versioned and then the unversioned, return if hit
+        for name in (f"{image_name}:{armory.__version__}", image_name):
```

Comment:
If image_name is something like `twosixarmory/pytorch:0.14.5.g24632d9`, it will look for `twosixarmory/pytorch:0.14.5.g24632d9:0.14.5.g24632d9`. However, this will succeed for `twosixarmory/pytorch`.

Reply:
This code is what translates `pytorch` to `twosixarmory/pytorch:0.14.5.g24632d9`, which is why you get the same error behavior for those two inputs.

https://github.com/shenshaw26/armory/blob/duo/armory/__main__.py#L82-L94

Reply:
My recommendation is that we should move all of the translation of `[twosixarmory/](pytorch|tf2|pytorch-deepspeech)[:<version>]` to `twosixarmory/(pytorch|tf2|pytorch-deepspeech):<version>` to `armory/__main__.py` and just have evaluator for attempting to download a specific image name. This will require adding functionality to `armory/__main__.run()` to modify the docker line in the config, though.

Reply:
This is now open for discussion in #1320